script/constants.lua

```diff
@@ -10,37 +11,37 @@ Constants.PLACER_TO_RAMP_SHIFT_BY_DIRECTION[defines.direction.east]  = {  0.5,
 Constants.PLACER_TO_RAMP_SHIFT_BY_DIRECTION[defines.direction.south] = { -0.5,  0.5 }
 Constants.PLACER_TO_RAMP_SHIFT_BY_DIRECTION[defines.direction.west]  = { -0.5, -0.5 }
 
+-- wall1
+Constants.PLACER_TO_WALL1_SHIFT_BY_DIRECTION = {}
```

Comment:
Wall items are the same, so we do not need to make new constant entry for each wall. We can put these shifts into a single array of tuples, like [ {direction, {xcoord, ycoord}}, {direction, {xcoord, ycoord}} ... ], then use it with for loop when building the tunnel.
Same goes with signals.
Also, how about putting walls/signals in global tunnel variable into array in case we need to add more?

Reply:
trying to find out how to store them in an array. I do not why, but it keeps failing to store the object in array.
Possible reason is deep/shallow copy. I am keeping eyes on that to solve this.